src/com/facebook/buck/go/GoTool.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2015-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.go;
+
+import com.facebook.buck.rules.BuildRule;
+import com.facebook.buck.rules.RuleKeyBuilder;
+import com.facebook.buck.rules.SourcePath;
+import com.facebook.buck.rules.SourcePathResolver;
+import com.facebook.buck.rules.Tool;
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableList;
+
+import java.nio.file.Path;
+
+public class GoTool implements Tool {
```

Comment:
Agreed. I can do a separate CR to add that, but since Tool is so widely used, I have a few choices:
 - Attempt to fix all the places that call getCommandPrefix to also pass-through getEnvironment. There's about 90 places where it's used, and this will likely be merge hell for you guys (or me).
 - Add the abstraction, but leave all the places that use them in a buggy state. Not great, but leaves a bunch of surprising tech debt.
 - Add environment variables as a new Tool builder that basically defers to `/usr/bin/env X=Y ...` to set the env variables. Easiest, but not particularly clean.

Thoughts?